src/libpmemobj/memops.c

```diff
@@ -80,6 +80,9 @@ struct operation_context {
 	struct ulog *ulog; /* pointer to the persistent ulog log */
 	size_t ulog_base_nbytes; /* available bytes in initial ulog log */
 	size_t ulog_capacity; /* sum of capacity, incl all next ulog logs */
+	size_t ulog_auto_reserve; /* allow or do not to auto ulog reservation */
+	size_t ulog_any_prealloc; /* set if any prealloc occured */
+	size_t ulog_first_prealloc; /* set if it is the first user prealloc */
```

Comment:
1) those 3 variables don't need to be so wide, bool or int is enough
2) it seems 2nd and 3rd variables has an opposite meaning, do we need both?